### ISSUE TITLE
Support setting StorageClass on individual GCS objects

### ISSUE THREAD

**Issue by daltschu22** [rclone/rclone#3043]
As an rclone GCS user, I would like to be able to set storage class for each object individually, separate from the bucket's current default class. 
As it is currently, during non server side copies, the StorageClass does not attempt to be set per object.
So for example, I could have a bucket with a default of `StorageClass = REGIONAL`, but I want to upload a series of files into the bucket with the `StorageClass = COLDLINE` defined per object, there is no way to do this.

GCS supports the option of setting an individual object storage class, different from the bucket class default. So I could have half my objects be Regional, and half Nearline in the same bucket for example.
Or my buckets default could be set to Regional, but I decide to upload any new objects as Coldline.
This is described here:
https://cloud.google.com/storage/docs/per-object-storage-class
https://cloud.google.com/storage/docs/storage-classes#bucket_vs_object_storage_class
The reason for this is generally so people can do tiering within the bucket.

The metadata field on the Object for that setting is:
```
    // StorageClass: Storage class of the object.
    StorageClass string `json:"storageClass,omitempty"`
```

From a previous thread:
> > I wonder if there should be another option added called object_storage_class to distinguish it from the bucket storage class?

> That would be easy enough to do, then you could set --gcs-object-storage-class when you wanted to do something different to what is set on the bucket.

>> Either that, or just go about setting the objects storage class individually based on the existing storage_class configuration setting.

> That would be easy enough too.
> I think the first idea is probably the most flexible, but I wonder whether it is overkill?

I think you are correct that adding a separate config seems like overkill. In my opinion it all comes down to how the user expects the application will function. 
Based on current documentation `The storage class to use when storing objects in Google Cloud Storage.` this seems misleading, given this issue. I can upload objects today, and they will be stored with a different class than what is defined in the config.

I took a very brief look at the AWS documentation (Im much less familiar with it) and it seems they ALSO support individual object StorageClasses. 
https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-class-intro.html#sc-howtoset

So its possible this change could apply to other systems beyond GCS?

**Comment by ncw** [comment]
Do you want to have a go at implementing this?  I'm happy to talk you through it.

**Comment by daltschu22** [comment]
Yes, absolutely. I will likely need some time to familiarize myself with Go and the existing codebase (I see you have docs already to help folks with this), but this is a good excuse to start learning Go!

Ill see how far I can get before I need help!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.54

Thank you @iiLaurens

Note you apply the fix with `--header-upload "x-goog-storage-class: class"`

It might be that this should be a --gcs-object-storage-class too but the above at least allows it to be set now.
